flake8_trio.py

```diff
@@ -36,66 +36,206 @@ def make_error(error: str, lineno: int, col: int, *args: Any, **kwargs: Any) ->
     return (lineno, col, error.format(*args, **kwargs), type(Plugin))
 
 
-def is_trio_call(node: ast.AST, *names: str) -> Optional[str]:
+class TrioScope:
+    def __init__(self, node: ast.Call, funcname: str, packagename: str):
+        self.node = node
+        self.funcname = funcname
+        self.packagename = packagename
+        self.variable_name: Optional[str] = None
+        self.shielded: bool = False
+        self.timeout: bool = False
+
+        if self.funcname == "CancelScope":
+            for kw in node.keywords:
+                # Only accepts constant values
+                if kw.arg == "shield" and isinstance(kw.value, ast.Constant):
+                    self.shielded = kw.value.value
+                # sets to True even if timeout is explicitly set to inf
+                if kw.arg == "deadline":
+                    self.timeout = True
+        else:
+            self.timeout = True
+
+    def __str__(self):
+        # Not supporting other ways of importing trio
+        # if self.packagename is None:
+        # return self.funcname
```

Comment:
This is reasonable, but we'll also want to lint on "some way of importing stuff from trio that doesn't have `trio.` at the start of each name".  Not a matter for this PR, but leave a comment on the issue?

Reply:
hmm, I specifically remember you mentioning in one of these pull requests that you didn't care to support anything but `import trio` -> `trio.funcname`. I had a version of this that was on it's way of working similar to how we're doing with context managers, and wouldn't be hard to reinstate that - at the risk of false positives.

Reply:
I mean that we should error on "from trio import" or "import trio as"!

Reply:
Aah, got it. That's an easy one :)